ESMF/GOCART2G_GridComp/CA2G_GridComp/CA2G_GridCompMod.F90

```diff
@@ -550,10 +550,13 @@ subroutine Initialize (GC, import, export, clock, RC)
     call ESMF_ConfigGetAttribute (universal_cfg, self%diag_MieTable(instance)%channels, &
                                   label= "aerosol_monochromatic_optics_wavelength_in_nm_from_LUT:", __RC__)
 
+!   Convert input wavelengths from nm to m for internal use
+    self%diag_MieTable(instance)%channels = self%diag_MieTable(instance)%channels * 1.0e-9
```

Comment:
This is confusing to me. I have a hard time seeing the need of performing this `nm`-to-`m` conversion multiple times (in different components) when this could be done once in the GOCART 2G gridded component and stored in `universal_cfg`, to be retrieved by all other components as needed in the proper units.

Reply:
I think the main issue is that we must do the conversion after the get from the resource file. Since above this we are getting `aerosol_monochromatic_optics_wavelength_in_nm_from_LUT` from `universal_cfg` into `self%diag_MieTable(instance)%channels`, we convert to meters after this. There is no point before this where `self%diag_MieTable(instance)%channels` is filled.

Reply:
@mathomp4 - I understand your point and generally agree with it. However, in this case, optical properties are not read from a configuration file associated with `universal_cfg`, but rather _set_ in `universal_cfg` by the GOCART 2G gridded component (`universal_cfg` is used similarly to an internal state):

https://github.com/GEOS-ESM/GOCART/blob/e3682659c00d317652d4743963a191a5e63ae67f/ESMF/GOCART2G_GridComp/GOCART2G_GridCompMod.F90#L156-L160

This is why I'm suggesting that we could perform the conversion only once before setting the above values in `universal_cfg`. I also understand that setting wavelengths in `m` for internal use will contrast with the updated configuration labels in `universal_cfg`, which explicitly include `nm` in their name.

Reply:
Maybe, but the label ***explicitly*** says `in_nm` and I cannot go against that. If a scientist wishes to make that change, okay, but for now, I obey the label.

Plus, what if a scientist in the future pulls from the config expecting it to be in `nm` and they get `m`?

Can you show @tclune or I a profile of UFS-Aerosols showing this multiplication is a tall-pole in your runs? I can't imagine it is that expensive. I'll try and dump the assembly code tomorrow to see, but I can't imagine it is that many instructions, especially seeing as we do this in the initialize method. Does UFS call the initialize method multiple times such that this is a high cost?

Reply:
@mathomp4 - I do understand all of your points and I'm totally fine with merging this PR as is. My suggestion was mostly driven by the desire to simplify the code base, which may help prevent bugs in the future.